tests-e2e/fixture/gitopsdeployment/fixture.go

```diff
@@ -70,3 +71,45 @@ func HaveSyncStatusCode(status managedgitopsv1alpha1.SyncStatusCode) matcher.Gom
 		return res
 	}, BeTrue())
 }
+
+// HaveResources checks if the .status.resources field of GitOpsDeployment have the required resources
+func HaveResources(resourceStatusList []managedgitopsv1alpha1.ResourceStatus) matcher.GomegaMatcher {
+	return WithTransform(func(gitopsDeployment managedgitopsv1alpha1.GitOpsDeployment) bool {
+		k8sClient, err := fixture.GetKubeClient()
+		if err != nil {
+			fmt.Println(k8sFixture.K8sClientError, err)
+			return false
+		}
+
+		err = k8sClient.Get(context.Background(), client.ObjectKeyFromObject(&gitopsDeployment), &gitopsDeployment)
+		if err != nil {
+			fmt.Println(k8sFixture.K8sClientError, err)
+			return false
+		}
+
+		// compare the slices irrespective of their order
+		resourceExists := false
+		existingResourceStatusList := gitopsDeployment.Status.Resources
+
+		if len(resourceStatusList) != len(existingResourceStatusList) {
+			fmt.Println("HaveResources:", resourceExists, "/ Expected:", resourceStatusList, "/ Actual:", gitopsDeployment.Status.Resources)
+			return false
+		}
+
+		for _, resourceStatus := range resourceStatusList {
```

Comment:
Do we care about the order of elements in the list?  I'm assuming that Argo CD could populate the Application's resources field in any order. Hence I'm comparing the objects by ignoring the order rather than using `reflect.DeepEqual`

Reply:
Agree, +1 for DeepEqual()